src/device/rpi3/date.go

```diff
@@ -0,0 +1,103 @@
+package rpi3
+
+// for computing the current time
+var startTime uint32
+var startTicks uint64
+
+//SetStartTime should be called exactly once, at boot time of a program that
+//is loaded by the bootloader.
+func SetStartTime(nowUnix uint32) {
```

Comment:
While undoubtely useful, I think setting the current time belongs in the runtime package. When done correctly, it will work for all targets at once.